### ISSUE TITLE
Election Polynomial should not use lists in internal structure

### ISSUE THREAD

**Issue by AddressXception** [microsoft/electionguard-python#84]
the `ElectionPolynomial` NamedTuple uses various types of lists, each of which is correlated to the others.  instead we should use a dictionary structure so that it is clear which values map together.

alternatively, we could use an internal type that includes the coefficient, it's commitment, and its proof and simply use a collection of the rich object type instead

**Comment by ddrm86** [comment]
I cannot figure out a dictionary structure for this that is not a bit cumbersome to read and use. Maybe a list of triplets? The triplets can be named tuples so each field can be accessed by its name.
Please assign this issue to me if you like my approach.

**Comment by AddressXception** [comment]
> I cannot figure out a dictionary structure for this that is not a bit cumbersome to read and use. Maybe a list of triplets? The triplets can be named tuples so each field can be accessed by its name.
> Please assign this issue to me if you like my approach.

yes, let's try it.  I was also struggling to come up with something that felt expressive and natural.  Let's see how this looks.

**Comment by ddrm86** [comment]
I tried something like this:

    class ElectionPolynomial(NamedTuple):
        """
        A polynomial defined by coefficients
    
        The 0-index coefficient is used for a secret key which can
        be discovered by a quorum of n guardians corresponding to n coefficients.
        """
        class Triplet(NamedTuple):
            coefficient: ElementModQ
            """The secret coefficients `a_ij` """
            coefficient_commitment: ElementModP
            """The public keys `K_ij`generated from secret coefficients"""
            coefficient_proof: SchnorrProof
            """A proof of possession of the private key for each secret coefficient"""
        
        triplets: List[Triplet]
    
but I am not happy about it:

- `Triplet` as an inner class is cumbersome to work with: `ep = ElectionPolynomial([ElectionPolynomial.Triplet(_, _, _)])` and it has to be exposed to client code.

- We lose the ability to get all the coefficients, commitments, or proofs easily. I am not sure how important is this, but it is used at least once in [test_verify_polynomial_coordinate](https://github.com/microsoft/electionguard-python/blob/daf40287fad0ad337bc1402acb69bfb430ac2fcd/tests/test_election_polynomial.py#L36)

Maybe we need a full-blown class instead of a NamedTuple?
